### ISSUE TITLE
Shellcheck doesn't always warn of unsafe array subscript expansions in arithmetic context

### ISSUE THREAD

**Issue by kerframil** [koalaman/shellcheck#1322]
As tested in shellcheck 0.5.0 and the online version, the following sample does not warn of the unsafe expansion of $bar:-

```shcipherfox mozilla addons
#!/bin/bash
declare -A foo
bar='$(echo "evil code" >&2)a'
(( foo[$bar]++ ))
declare -p foo
```

Here's what the program itself prints:

```
evil code
evil code
declare -A foo=([a]="1" )
```

Here's what shellcheck reports:-

```
bar='$(echo "evil code" >&2)a'
    ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
```

What would be preferable is for shellcheck to also report that the expansion of the subscript is unsafe due to the arithmetic context. For reference, this is the text from greybot's current `$subscript` entry:-

> In an arithmetic context, never reference an array subscript dynamically as foo[$bar]. Instead, write foo[bar] for indexed arrays, and any of foo[\\$bar], foo['$bar'] or ${foo[$bar]} for associative arrays. Otherwise, the expansion of bar is repeated and is thus vulnerable to code injection.

Perhaps the scope of SC2004 could be expanded to cover this.

**Comment by nepella** [comment]
The same goes for using the `[[ -v ... ]]` test:

```sh
#!/bin/bash
declare -A foo
bar='$(echo "Gotcha!" >&2)'
[[ -v foo[$bar] ]]
```

**Comment by terrorbyte** [comment]
This came up in the real world again this week with a RCE in Ivanti EPMM: https://labs.watchtowr.com/someone-knows-bash-far-too-well-and-we-love-it-ivanti-epmm-pre-auth-rces-cve-2026-1281-cve-2026-1340/

**Comment by wileyhy** [comment]
Virustotal.com:

"1/94 security vendor flagged this URL as malicious"

**Comment by terrorbyte** [comment]
> Virustotal.com:
> 
> "1/94 security vendor flagged this URL as malicious"
> […](#)

Yes, it's an exploit writeup...... by a well established research group, that is not exactly shocking that a random vendor decided it was bad or flags it as a not allowed link.
